### ISSUE TITLE
Data Augmentation and hyper-parameters setting

### ISSUE THREAD

**Issue by ehsanpi** [ultralytics/ultralytics#14613]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I'm truly amazed by the power of YOLOv8 and it's super ease of implementation. WOW! I'm training a model for object detection on my image dataset of 3500 images of 2048x2048 pixels. I've checked the relevant post in "https://github.com/ultralytics/ultralytics/issues/2412",
 and proceeded accordingly. However, it has rather a negative impact on the results! I double-checked the labelled data in the Model_Outputs for both training and validation datasets and inspected if the bounding boxes, especially once they are rotated or rescaled (by passing extra variables to the command line for training the model) are displayed correctly around the object of interest. 
 
I reckon it is not the YOLO default and I need to change the hyper-parameters in particular, learning rates. For example, any learning rate but 0.01 for both lr0 and lrf is not working. I even tried the larger batch size of 36 (distributing amongst 4 GPUs) but still not as good as without data augmentation. 

Believe it or not, I've spent 840 USD thus far on AWS to improve the accuracy with no success. Could you please advise what is best to do or if there would be any specific heads up? Following is the code that includes the hyper-parameters for training.

Many thanks for sharing your thoughts in advance!

### Minimal Reproducible Example

#Following is the command for training the model:
!yolo task=detect mode=train  model='yolov8n.pt'  epochs=250 warmup_epochs=5 lr0=0.05  lrf=0.001 batch=36 imgsz='2048,2048' data='.../config.yaml'  project='.../Model_Outputs/'  name='name' save_period=1 device=0,1,2,3  patience=20   augment=True  degrees= 360  flipud=1 fliplr=1  copy_paste=1  optimizer ='SGD'      

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
How small are your objects? Unless they are really small, that large resolution is a waste. `640` would be enough and it's better to spend that extra compute training a larger model (m, l, or x).

lrf stands for learning rate factor and not final learning rate. The final lr would be `lr0 * lrf`, 0.00005 in your case.

Setting fliplr or flipud or any of the augmentation to 1.0 defeats the point of the augmentation. Because it means all the images will be flipped vertically and horizontally **always**. So your model is only trained on upside down and mirrored images. You should be using 0.5, so that it does it only 50% of the time.

**Comment by ehsanpi** [comment]
> How small are your objects? Unless they are really small, that large resolution is a waste. `640` would be enough and it's better to spend that extra compute training a larger model (m, l, or x).
> 
> lrf stands for learning rate factor and not final learning rate. The final lr would be `lr0 * lrf`, 0.00005 in your case.
> 
> Setting fliplr or flipud or any of the augmentation to 1.0 defeats the point of the augmentation. Because it means all the images will be flipped vertically and horizontally **always**. So your model is only trained on upside down and mirrored images. You should be using 0.5, so that it does it only 50% of the time.

**Comment by ehsanpi** [comment]
@Y-T-G Thank you so much for your thorough and swift response! I really though "lrf" is the final learning rate and now thanks to you I'm aware of this fatal mistake! Same about the flip! By the same token, I guess setting degrees=360 would also defeat the augmentation objective, right? and should be set to a max of 180 degrees (unsuccessful to pass either way of both clockwise and counter-clockwise rotation: [-180 to +180] degrees rotation to the command line).

To briefly describe the details, I'm trying to detect water bodies in an aerial image which remarkably vary in size, shape and visual appearance. I need to break the large image into smaller image tiles. However, the best I could get have achieved was a "mAP50 = 0.783", of course passing no augmentation argument to keep the default values.

In terms of down-sampling to 640x640 pixels, I have observed that 2048x2048 pixel resolution delivers a higher accuracy against 512x512 pixels resolution. Water bodies aren't smaller than 10 meters in diameter and the image resolution is 0.15 meters per pixel (so objects are larger than 70x70 pixels in a 2048x2048 image tile). Please see attached a sample of image tiles and objects used for training.
![image](https://github.com/user-attachments/assets/81747b0c-e224-4bfc-b541-3cb705ac6add)

**Comment by Y-T-G** [comment]
The gain you get from going from n to m, l or x is much better than higher resolution. In your case, the water bodies doesn't seem that small, especially requiring 2560. 1280 should suffice at most.

`degrees: 180` does both ±180 and anything it between, so it's essentially 360.

**Comment by Y-T-G** [comment]
Although, you should probably use some sort of super-sampling based downscaling algorithm and downscale it to a nearby resolution like 768p (if using 640 imgsz) prior to training to prevent those aliasing artefacts.
